scripts/update-versions.ps1

```diff
@@ -6,7 +6,7 @@ function Get-LatestNugetVersion {
 
     $url = "https://api.nuget.org/v3/registration5-gz-semver2/$($packageId.ToLowerInvariant())/index.json"
     $response = Invoke-RestMethod -Uri $url -Method Get
-    $response.items[0].upper
+    $response.items[$response.items.Count - 1].upper
```

Comment:
should we do client side ordering to ensure we grab the correct release?

Reply:
I don't think we need to go that far here. 
What nuget.org tries to do is group the items in groups(pages) with maximum sizes of 64, so as we've crossed 64 releases, we need to look at the `upper`(highest version) at the next page.
So essentially, the logic here is to pick the upper (latest version) in the last page.  

https://learn.microsoft.com/en-us/nuget/api/registration-base-url-resource#registration-pages-and-leaves

Reply:
they do paging within a single response? I'm not sure what lead to that decision but that's not a great developer experience...
Anyway, from their public documentation, nothing guarantees the pages will actually be ordered.
It'd probably be safer to collect all the upper bounds client side, order them, and take the highest one.

Something like this I guess

```powershell
$version = $response.items | Select-Object -ExpandProperty upper | % { [System.Management.Automation.SemanticVersion]$_ } | sort-object | Select-Object -Last 1
$version.ToString()
```

Reply:
No worries. Will update the PRs in a bit.

Reply:
Resolved via https://github.com/microsoft/kiota/pull/4827